src/app_service/service/metrics/service.nim

```diff
@@ -25,48 +38,44 @@ QtObject:
     result.QObject.setup
     result.threadpool = threadpool
 
-    signalConnect(singletonInstance.globalEvents, "addCentralizedMetric(QString, QString)",
-      result, "addCentralizedMetric(QString, QString)", 2)
+    signalConnect(singletonInstance.globalEvents, "addCentralizedMetricIfEnabled(QString, QString)",
+      result, "addCentralizedMetricIfEnabled(QString, QString)", 2)
 
   # eventValueJson is a json string
-  proc addCentralizedMetric*(self: MetricsService, eventName: string, eventValueJson: string) {.slot.} =
-    let arg = AsyncAddCentralizedMetricTaskArg(
-      tptr: asyncAddCentralizedMetricTask,
+  proc addCentralizedMetricIfEnabled*(self: MetricsService, eventName: string, eventValueJson: string) {.slot.} =
+    debug "Add metric if enabled for ", eventName, eventValueJson
+    let arg = AsyncAddCentralizedMetricIfEnabledTaskArg(
+      tptr: asyncAddCentralizedMetricIfEnabledTask,
       vptr: cast[ByteAddress](self.vptr),
-      slot: "onCentralizedMetricAdded",
+      slot: "onCentralizedMetricAddedIdEnabled",
       eventName: eventName,
       eventValueJson: eventValueJson,
     )
     self.threadpool.start(arg)
 
-  proc onCentralizedMetricAdded*(self: MetricsService, response: string) {.slot.} =
+  proc onCentralizedMetricAddedIdEnabled*(self: MetricsService, response: string) {.slot.} =
     try:
       let responseObj = response.parseJson
       let errorString = responseObj{"error"}.getStr()
       if errorString != "":
-        error "onCentralizedMetricAdded", error=errorString
+        error "onCentralizedMetricAddedIdEnabled", error=errorString
+        return
+      
+      if responseObj{"metricsDisabled"}.getBool:
+        debug "metrics collection is turned off"
```

Comment:
I also wondered if it would be annoying. Since I was logging the `Add metric if enabled for`, I thought it would be important to log that we didn't actually collect the metric.

But I think I can just remove those two logs and only log when we do collect